### ISSUE TITLE
Handling the new share-or-own-process services

### ISSUE THREAD

**Issue by angelsl** [processhacker2/processhacker#120]
Since [build 14942](https://blogs.windows.com/windowsexperience/2016/10/07/announcing-windows-10-insider-preview-build-14942-for-pc/) services with start type `SERVICE_WIN32_SHARE_PROCESS` are run in their own `svchost.exe` unless `SvcHostSplitDisable` (`REG_DWORD`) is set to 1 in the service's registry key.

When a running share-process service that is actually run under its own process is queried e.g. via `EnumServicesStatusEx`, `dwServiceType` is returned as `SERVICE_WIN32` which is

```c
#define SERVICE_WIN32 (SERVICE_WIN32_OWN_PROCESS | SERVICE_WIN32_SHARE_PROCESS)
```

When such a service is not running `dwServiceType` is returned as `SERVICE_WIN32_SHARE_PROCESS`.

`SERVICE_WIN32` is not accepted as a valid value of `dwStartType` by `CreateService` or `ChangeServiceConfig`. (Not sure what MS was thinking with this one...)

I'm not really sure of any nice way to handle this; we obviously cannot simply add `SERVICE_WIN32` as a new value in `PhpServiceTypePairs` because it's never accepted as an argument.

**Comment by samlive9090** [comment]
Hello,

Can You Add These Things In Service Type:-

1) SERVICE_PKG_SERVICE            0x00000200

2) SERVICE_WIN32                 
 (SERVICE_WIN32_OWN_PROCESS | SERVICE_WIN32_SHARE_PROCESS)

3) Enter Custom Value For Setting Service Type.

//
// Service Types (Bit Mask)
//
#define SERVICE_KERNEL_DRIVER          0x00000001
#define SERVICE_FILE_SYSTEM_DRIVER     0x00000002
#define SERVICE_ADAPTER                0x00000004
#define SERVICE_RECOGNIZER_DRIVER      0x00000008

#define SERVICE_DRIVER                 (SERVICE_KERNEL_DRIVER | \
                                        SERVICE_FILE_SYSTEM_DRIVER | \
                                        SERVICE_RECOGNIZER_DRIVER)

#define SERVICE_WIN32_OWN_PROCESS      0x00000010
#define SERVICE_WIN32_SHARE_PROCESS    0x00000020
#define SERVICE_WIN32                  (SERVICE_WIN32_OWN_PROCESS | \
                                        SERVICE_WIN32_SHARE_PROCESS)

#define SERVICE_USER_SERVICE           0x00000040
#define SERVICE_USERSERVICE_INSTANCE   0x00000080

#define SERVICE_USER_SHARE_PROCESS     (SERVICE_USER_SERVICE | \
                                        SERVICE_WIN32_SHARE_PROCESS)
#define SERVICE_USER_OWN_PROCESS       (SERVICE_USER_SERVICE | \
                                        SERVICE_WIN32_OWN_PROCESS)

#define SERVICE_INTERACTIVE_PROCESS    0x00000100
#define SERVICE_PKG_SERVICE            0x00000200

#define SERVICE_TYPE_ALL               (SERVICE_WIN32  | \
                                        SERVICE_ADAPTER | \
                                        SERVICE_DRIVER  | \
                                        SERVICE_INTERACTIVE_PROCESS | \
                                        SERVICE_USER_SERVICE | \
                                        SERVICE_USERSERVICE_INSTANCE | \
                                        SERVICE_PKG_SERVICE)











[wdm.zip](https://github.com/processhacker2/processhacker2/files/1063450/wdm.zip)
[wdm.txt](https://github.com/processhacker2/processhacker2/files/1063452/wdm.txt)

**Comment by dmex** [comment]
>SERVICE_WIN32 is not accepted as a valid value of dwStartType by CreateService or ChangeServiceConfig. (Not sure what MS was thinking with this one...)

Fixed in [#7b5570a2](https://github.com/processhacker2/processhacker/commit/7b5570a25048e3618d1f1ffa30f3a86f56cb8303)
